src/lib/turtle.js

```diff
@@ -849,9 +849,39 @@ function generateTurtleModule(_target) {
         };
 
         proto.$speed = function(speed) {
-            if (speed !== undefined) {
-                this._speed          = Math.max(0, Math.min(1000, speed));
-                this._computed_speed = Math.max(0, speed * 2 - 1);
+            if (typeof(speed) !== "undefined") {
+                if (typeof(speed) !== "string" && typeof(speed) !== "number") {
+                    throw new Sk.builtin.TypeError("speed expected a string or number");
+                }
+                if (typeof(speed) === "string") {
+                    if (speed === "fastest") {
+                        speed = 0;
+                    } else if (speed === "fast") {
+                        speed = 10;
+                    } else if (speed === "normal") {
+                        speed = 6;
+                    } else if (speed === "slow") {
+                        speed = 3;
+                    } else if (speed === "slowest") {
+                        speed = 1;
+                    } else {
+                        speed = this._speed;
```

Comment:
in python if the speed string is not one of these values then an error will be raised (the internal implementation of python will try to to compare int and str)

perhaps something like

```javascript
var speeds = {fastest: 0, fast: 10, ... }
if (speed in speeds) {
   speed = speeds[speed]
}
if (typeof speed !== "number") {
  throw ...
}
```

Reply:
I added a new commit for this one, the exception is a little tricky. In Python it will throw `TypeError: '<' not supported between instances of 'float' and 'str'` if the speed string is not one of "fastest", "fast", "normal", "slow", "slowest" when it tries to run code `elif 0.5 < speed < 10.5:`; Here I threw a TypeError to indicate it is not one of the five value, but technically I guess it should be "ValaueError" ? So TypeError here is just trying to make it throw the same exception like Python does. Thoughts?
```
            if (typeof(speed) !== "number") {
                if (typeof(speed) === "string") {
                    const valid_speeds = Object.keys(speeds).join(", ");
                    throw new Sk.builtin.TypeError("speed string expected one of " + valid_speeds);
                }
                throw new Sk.builtin.TypeError("speed expected a string or number");
            }
```